src/auth/authService.ts

```diff
@@ -27,27 +35,67 @@ export class AuthService {
         name,
         surname,
         email,
-        hashpassword
+        hashpassword,
+        code
       );
+      await this.mail.sendMail(user.rows[0].id, email, code);
       return user;
     }
   }
 
   async loginUser(email: string, password: string) {
     let hashpassword = createHash("sha256").update(password).digest("hex");
     let userLogin = await this.repository.findUserByEmail(email);
-    if (userLogin.password === hashpassword) {
-      return {
-        token: jsonwebtoken.sign(
-          {
-            id: userLogin.id,
-            access_type: userLogin.access_type,
-          },
-          "secret"
-        ),
-      };
+    const isVerified = await this.repository.isVerified(userLogin.id);
+    if (isVerified) {
+      if (userLogin.password === hashpassword) {
+        return {
+          token: jsonwebtoken.sign(
+            {
+              id: userLogin.id,
+              access_type: userLogin.access_type,
+            },
+            "secret"
+          ),
+        };
+      } else {
+        throw new WrongCredentials();
+      }
     } else {
-      throw new WrongCredentials();
+      throw new EmailNotVerifies();
+    }
+  }
+
+  async verifyEmail(code: number, user_id: number){
+    let checkCode = await this.repository.getCode(user_id);
+    if(checkCode === code){
```

Comment:
its looks like better, then there is less main words (if else etc).
```js
if(checkCode !== code) throw new WrongCode();
// here we dont need else anymore because u will throw error before u reach this code base
return this.repository.changeEmailIsVerified(user_id);

```